### DISCUSSION TITLE
Is there a reason for @Mapping do not have an @Inherited on it ?

### DISCUSSION THREAD

**Discussion by cvgaviao** [mapstruct/mapstruct#3712] [Q&A]
I tried to create a generic mapper interface today and it worked well.

```
interface BaseMapper<E extends BaseEntity, D extends BaseDto>{

E toEntity(D pDto);

@Mapping(target = "createdBy", ignore = true)
@Mapping(target = "createdOn", ignore = true)
@Mapping(target = "optLock", ignore = true)
@Mapping(target = "updatedBy", ignore = true)
@Mapping(target = "updatedOn", ignore = true)
D toDto(E pEntity);

```
```
interface FooEntityMapper extends BaseMapper<Foo, FooDto>{

}
```
I have created FooEntityMapper interface extending BaseMapper and it worked well... until I tried to override toDto to add another @Mapping.

Surprisingly @Mapping do not contain @Inherited. 

So, to resolve my problem I need to copy all mappings from BaseMapper to the  FooEntityMapper's toDto method.

So, I'm wondering... is there a  reason for @Mapping do not be inheritable ?

**Comment by filiphr** [comment]
@cvgaviao I don't think that the use of `@Inherited` fits for what you are looking for. In any case, I think what you are looking for is #3403

**Reply by cvgaviao** [reply]
Hello @filiphr , thanks for your time.

Indeed the suggestion of change the name of the method on the child interface and use `@InheritConfiguration` should work. I will try it today.

But I also think that it is more verbose and less intuitive than just override the method (Ide does that with a click) and add the desired @Mapping or override the ones you wanted, and then just use what the java annotation and the processor api already gives us, the array of annotations of a method in the hierarchy (through the @inhereted support).

**Reply by cvgaviao** [reply]
Btw, I cloned the project yesterday and added the @Inhereted in all mapping annotations and run the build just to see if all the tests would pass. and they did.

But I don't know yet how to create a test for this inheritance scenario :)

**Reply by cvgaviao** [reply]
I tested your suggestion and seems not to be working as expected. 
I added a mapping to ignore the attribute teams in mapUpdateDtoToEntityOverriding  but the warning still there.

<img width="796" alt="image" src="https://github.com/user-attachments/assets/fd005644-1dc3-4899-ae2e-04d3a2a7d1fd">

**Reply by filiphr** [reply]
@cvgaviao what I meant with referencing #3403 was more of a reference to the potential enhancement to support this e.g. ` @BeanMapping(inheritSuperMappings = true)`

>Btw, I cloned the project yesterday and added the @Inhereted in all mapping annotations and run the build just to see if all the tests would pass. and they did.

I was not expecting tests to fail, I don't think that we have an exact test which tests the scenario you need.

>I tested your suggestion and seems not to be working as expected.

The example in the issue is different to your example. For your example to work you'll need to override the parrent methods and delegate to the new method.

**Reply by cvgaviao** [reply]
> @cvgaviao what I meant with referencing #3403 was more of a reference to the potential enhancement to support this e.g. ` @BeanMapping(inheritSuperMappings = true)`
> 
> > Btw, I cloned the project yesterday and added the @Inhereted in all mapping annotations and run the build just to see if all the tests would pass. and they did.
> 
> I was not expecting tests to fail, I don't think that we have an exact test which tests the scenario you need.
> 
Ok, but my intention was to verify that any existent tests/behavior broke after changing the mappings annotation to use @Inhered .

> > I tested your suggestion and seems not to be working as expected.
> 
> The example in the issue is different to your example. For your example to work you'll need to override the parrent methods and delegate to the new method.

The difference is that I included more parameters in the generic interface and changed its name :)

Well, can you please elaborate on this last suggestion ?  If I understand what you wrote, overriding the parent I would loose all the mappings and I would need to copy all of the them to new method, wouldn't I?

**Reply by filiphr** [reply]
>Ok, but my intention was to verify that any existent tests/behavior broke after changing the mappings annotation to use `@Inherited`.

The thing is that I do not believe that adding it would achieve what you think it'll achieve.

>If I understand what you wrote, overriding the parent I would loose all the mappings and I would need to copy all of the them to new method, wouldn't I?

yes my bad, I got confused with something else. In any case what you are asking for is what is being asked in #3403

**Comment by cvgaviao** [comment]
Hey @filiphr, could you please point me to the class(es) where Mapstruct builds its own model objects on its processor? 

I mean, where it loads / reads all `java.lang.annotation.Annotation` from a mapper interface / class / methods?
